apps/src/studio/utils/commandQueue.js

```diff
@@ -0,0 +1,39 @@
+import { singleton as studioApp } from '@cdo/apps/StudioApp';
+import api from '../api';
+
+/**
+ * Execute the code for all of the event handlers that match an event name
+ * @param {string} name Name of the handler we want to call
+ * @param {boolean} allowQueueExtension When true, we allow additional cmds to
+ *  be appended to the queue
+ * @param {Array} extraArgs Additional arguments passed into the virtual JS
+ *  machine for consumption by the student's event-handling code.
+ */
+export function callHandler(name, allowQueueExtension, extraArgs = []) {
+  Studio.eventHandlers.forEach(function (handler) {
+    if (studioApp().isUsingBlockly()) {
```

Comment:
I'd start by breaking out `callBlocklyHandler` and `callDropletHandler` so the `callHandler` function only adds the decision about which to use.  Then you might be able to bind just the needed one in studio.js at initialization (or as soon as we know `editCode` and pull the decision out of this file.